### PR TITLE
Add stack-haddock-upload script to upload docs to hackage.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/toktok/ci-tools/20)

### CODE REVIEW

File: bin/stack-haddock-upload [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# Author: Dimitri Sabadie <dimitri.sabadie@gmail.com>
+# 2015
+#
+# Retrieved from: https://github.com/phaazon/stack-haddock-upload/blob/53e22b616810c07b9516f42ef57365eb25525db4/stack-haddock-upload
```
Comment (robinlinden) [inline bin/stack-haddock-upload:6]: There doesn't seem to be a licence associated with the repository?

Comment (iphydf) [reply bin/stack-haddock-upload:6]: https://github.com/phaazon/stack-haddock-upload/issues/1

Yeah.. I don't know, this is tiny code, but still yes.

Comment (phaazon) [reply bin/stack-haddock-upload:6]: Just use it. :) I’ll put a WTFPL on it because I really don’t care. :)

Comment (phaazon) [reply bin/stack-haddock-upload:6]: Actually, I made it BSD 0-Clause. Have fun peeps.

Comment (robinlinden) [reply bin/stack-haddock-upload:6]: Awesome, thank you! :)

**Pull request merged**